### ISSUE TITLE
[Python] Implement new Concept Exercise: lists

### ISSUE THREAD

**Issue by aldraco** [exercism/v3#859]
This issue describes how to implement the `lists` concept exercise for the Python track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the basics of the `List` ([sequence type](https://docs.python.org/3/library/stdtypes.html#sequence-types-list-tuple-range)) data type in Python. We will accomplish this by requiring the student to create lists via various methods, iterate through a list via loop, access items in the list at a certain index, and form/access slices of the list.

## Learning objectives

 - Create a `list` via constructor (`list()`) & literal (**`[]`**)
-  Combine two or more lists by concatenation via **`+`**
-  Check for an items membership/absence in a list using `in` and/or `not in` 
 - Access items in a list via index (`bracket notation`)
 - Access a range of items in a list via list slicing (**`[start:stop:step]`**)
 - Iterate through a list using `for item in`
 - Understand that `lists` can store mixed/any data types
- Understand that `lists` are _mutable_.  Assigning a new value to a list index will change the value at that index. 

## Out of scope

 - List Comprehensions (_these will be covered in their own concept exercise_)
 - List methods (`copy()`, `append()`,  `sort()` etc)
 - How builtin functions relate to lists (_`len()`, `max()`, `min()`, `count()`, `sorted()`_)
 - Construction of complex or nested `lists` (i.e a list of `dicts`, a `list` of `lists`, a `list` of `tuples`)
 - Consequences and considerations of mutability

## Concepts

 - `lists`
 -  indexing 
 - slicing
 - concatenation
 - iteration

## Prerequisites

- [builtin-types](https://github.com/exercism/v3/tree/master/languages/python/reference/concepts/builtin_types)
- [data-types](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/data_structures.md)

## Resources to refer to

[Python lists doc](https://docs.python.org/3/tutorial/datastructures.html)

### Hints

Hints should link to the `list` [instance methods doc page](https://docs.python.org/3/tutorial/datastructures.html#more-on-lists).

### After

After, the student can explore the list comprehension pattern, even though it will be taught separately. This is also a good time to reference some builtin methods like `len()`, or `max()` - not as a teaching item, but to expose the student to their existence and build familiarity.

## Representer

No changes required.

## Analyzer

No changes required.

## Implementing

Tests should be written using `unittest.TestCase`, and the test file named `lists_basic_test.py`.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

## Edits
- edited for updated naming by @yawpitch
- edited to add additional learning objectives by @BethanyG.
- edited to clarify out of scope language by @BethanyG 
- edited After section wording.  @BethanyG

**Comment by BethanyG** [comment]
@itamargal will be taking this on.  🌟 😄

**Comment by BethanyG** [comment]
This was closed by #1583 and  #1932
